enterprise/server/scheduling/priority_task_scheduler/priority_task_scheduler.go

```diff
@@ -395,26 +416,49 @@ func (q *PriorityTaskScheduler) stats() string {
 		len(q.activeTaskCancelFuncs), q.q.Len())
 }
 
-func (q *PriorityTaskScheduler) canFitAnotherTask(res *scpb.EnqueueTaskReservationRequest) bool {
-	// Only ever run as many sized tasks as we have memory for.
-	knownRAMremaining := q.ramBytesCapacity - q.ramBytesUsed
-	knownCPUremaining := q.cpuMillisCapacity - q.cpuMillisUsed
-	willFit := knownRAMremaining >= res.GetTaskSize().GetEstimatedMemoryBytes() && knownCPUremaining >= res.GetTaskSize().GetEstimatedMilliCpu()
+func (q *PriorityTaskScheduler) canFitTask(res *scpb.EnqueueTaskReservationRequest) bool {
+	// If we're running in exclusiveTaskScheduling mode, only ever allow one
+	// task to run at a time. Otherwise fall through to the logic below.
+	if q.exclusiveTaskScheduling && len(q.activeTaskCancelFuncs) >= 1 {
+		return false
+	}
+
+	size := res.GetTaskSize()
+
+	availableRAM := q.ramBytesCapacity - q.ramBytesUsed
+	if size.GetEstimatedMemoryBytes() > availableRAM {
+		return false
+	}
 
-	// If we're running in exclusiveTaskScheduling mode, only ever allow one task to run at
-	// a time. Otherwise fall through to the logic below.
-	if willFit && q.exclusiveTaskScheduling && len(q.activeTaskCancelFuncs) >= 1 {
+	availableCPU := q.cpuMillisCapacity - q.cpuMillisUsed
+	if size.GetEstimatedMilliCpu() > availableCPU {
 		return false
 	}
 
-	if willFit {
-		if res.GetTaskSize().GetEstimatedMemoryBytes() == 0 {
-			log.CtxWarningf(q.rootContext, "Scheduling another unknown size task. THIS SHOULD NOT HAPPEN! res: %+v", res)
-		} else {
-			log.CtxInfof(q.rootContext, "Scheduling another task of size: %+v", res.GetTaskSize())
+	for _, r := range size.GetCustomResources() {
+		used, ok := q.customResourcesUsed[r.GetName()]
+		if !ok {
+			// The scheduler should never send us tasks that require resources
+			// we haven't set up in the config.
+			alert.UnexpectedEvent("missing_custom_resource", "Task requested custom resource %q which is not configured for this executor", r.GetName())
```

Comment:
How is this going to be configured on the client side? Mostly want to make sure that if clients are experimenting / have typos in their repos then we're not getting alerted.

Reply:
The task shouldn't make it this far if it asks for custom resources that the executor doesn't have, due to the logic [here](https://github.com/buildbuddy-io/buildbuddy/pull/6851/files#diff-fe896f3f44295b27c004e50a48cca5068273a3df48b98059bc62decb3c6df0caR504)